src/main/java/racingcar/RacingGameController.java

```diff
@@ -0,0 +1,49 @@
+package racingcar;
+
+import racingcar.domain.RacingManager;
+import racingcar.domain.CarNameParser;
+import racingcar.domain.numbergenerator.RandomNumberGenerator;
+import racingcar.view.InputView;
+import racingcar.view.OutputView;
+
+public class RacingGameController {
+    private final RandomNumberGenerator randomNumberGenerator;
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public RacingGameController(RandomNumberGenerator randomNumberGenerator) {
+        this.randomNumberGenerator = randomNumberGenerator;
+        this.inputView = new InputView();
+        this.outputView = new OutputView();
+    }
+
+    public void start(){
+        RacingManager racingManager = setUpManager();
+        executeGame(racingManager);
+        showGameResult(racingManager);
+    }
+
+    private RacingManager setUpManager() {
+        outputView.printCarNamesInputMessage();
+        String carNames = inputView.inputCarNames();
+        outputView.printAttemptsInputMessage();
+        int attempts = inputView.inputAttempts();
+        return new RacingManager(new CarNameParser().parse(carNames), attempts, randomNumberGenerator);
+    }
+
+    private void executeGame(RacingManager racingManager) {
+        outputView.printAttemptResultStartMessage();
+        do {
+            racingManager.doAttempt();
+            showAttemptResult(racingManager);
+        } while (!racingManager.isRaceEnd());
```

Comment:
do-while로 선택하신 이유가 있을까요? while문이 아닌 이유가 있는지 여쭤봐요!!

Reply:
while문으로 하면 이런식이라

```suggestion
     boolean continueRace = false;
     while (continueRace) {
          racingManager.doAttempt();
          showAttemptResult(racingManager);
          continueRace = racingManager.isRaceEnd()
      }
```

무조건 한번은 실행해향하는 구문이니까 do while문이 훨씬 깔끔하게 되더라구요 ㅎㅎ